h2/src/main/org/h2/command/Parser.java

```diff
@@ -5179,17 +5181,20 @@ else if(readIf("CREATE")) {
                     WITH_STATEMENT_SUPPORTS_LIMITED_STATEMENTS);
         }
 
-        // clean up temp views starting with last to first (in case of
-        // dependencies)
-        Collections.reverse(viewsCreated);
-        p.setCteCleanups(viewsCreated);
+        // clean up temporary views starting with last to first (in case of
+        // dependencies) - but only if they are not persistent
+        if(!isPersistent){
+            Collections.reverse(viewsCreated);
+            p.setCteCleanups(viewsCreated);
+        }
```

Comment:
if the views are inter-dependent you might need a loop to clean these up (which we already do in a couple of other places)

Reply:
These are the end-of-command "delayed" view cleanups - which don't happen in the persistent mode.